clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleInternalLoadBalancerCachingAgent.groovy

```diff
@@ -182,7 +182,13 @@ class GoogleInternalLoadBalancerCachingAgent extends AbstractGoogleLoadBalancerC
 
       def backendServiceName = Utils.getLocalName(forwardingRule.backendService)
       BackendService backendService = projectRegionBackendServices?.find { BackendService bs -> bs.getName() == backendServiceName }
-      handleBackendService(backendService, newLoadBalancer, projectHttpHealthChecks, projectHttpsHealthChecks, projectHealthChecks, groupHealthRequest)
+      if (backendService == null) {
+        log.warn("Failed to read a component of subject ${newLoadBalancer.name}. Could not find BackendService ${backendServiceName}.\n"
```

Comment:
Are there legitimate uses of GCP that can cause this?

Reply:
In that case I would include in the warning that this is simply in a format not expected/deployed by Spinnaker

Reply:
SGTM.

Reply:
We only call `cacheRemainderOfLoadBalancerResourceGraph` if the fowarding rule has a non-null backend service (https://github.com/spinnaker/clouddriver/blob/52d79d1a01e8fb9cbd6c65eb55f82b955f89477b/clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleInternalLoadBalancerCachingAgent.groovy#L161) so the condition `backendService == null` means that the load balancer defined a back-end service but we couldn't find it.

For TCP load balancers, we're looking up a back-end on a Target TCP Proxy, which is required to have a back-end---so again, a failure to get the back-end means that it existed but we couldn't find it.